### PR TITLE
Add skeleton for python bindings

### PR DESCRIPTION
**Pull request by GiulioRomualdi**
This pull request introduces Python bindings for the Biomechanical Analysis Framework, along with several supporting CMake scripts and configurations. The most important changes include adding a new `bindings` subdirectory, creating CMake functions for Python module integration, and setting up the necessary dependencies and configurations for building and installing the Python bindings.

I also created the IK folder (empty) for now. The PR also adds the support for the idyntree / swig casting

### CODE REVIEW

**Comment by GiulioRomualdi** [comment]
After some fixes I was able to load the bindings

```python
Python 3.10.12 (main, Sep 11 2024, 15:47:36) [GCC 11.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import biomechanical_analysis_framework as baf
>>> baf.ik
<module 'biomechanical_analysis_framework.bindings.ik'>
```

File: bindings/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        string(STRIP ${_PYTHON_INSTDIR} _PYTHON_INSTDIR_CLEAN)
+        set(FRAMEWORK_PYTHON_INSTALL_DIR ${_PYTHON_INSTDIR_CLEAN})
+      endif()
+    endif()
+    set(PYTHON_INSTDIR ${FRAMEWORK_PYTHON_INSTALL_DIR}/biomechanical_analysis_framework)
+
+
+    # Folder of the Python package within the build tree.
+    # It is used for the Python tests.
+    set(BLF_PYTHON_PACKAGE "${CMAKE_BINARY_DIR}/biomechanical_analysis_framework")
```
Comment (traversaro) [inline bindings/CMakeLists.txt:47]:
```suggestion
    set(BAF_PYTHON_PACKAGE "${CMAKE_BINARY_DIR}/biomechanical_analysis_framework")
```

**Comment by traversaro** [comment]
There are a few blf/baf instances that need to be changed.

**Comment by GiulioRomualdi** [comment]
> There are a few blf/baf instances that need to be changed.

Done :)

**Review by davidegorbani — APPROVED** [review]
Thank you @GiulioRomualdi!

**Review by evelyd — APPROVED** [review]
The import works for me, thanks @GiulioRomualdi :)

**Pull request merged**